asf_core_data/pipeline/preprocessing/feature_engineering.py

```diff
@@ -433,6 +459,59 @@ def get_building_entry_feature(df, feature):
     return df
 
 
+def enhance_construction_age_band(
+    construction_age_band: str,
+    transaction_type: str,
+    inspection_date: str,
+    country: str,
+) -> str:
+    """
+    Enhances construction age band by replacing unknown when transaction type is new dwelling and
+    inspection date is known.
+
+    Args:
+        construction_age_band: property's construction age band
+        transaction_type: transaction type (e.g. new dwelling)
+        inspection_date: inspection date as a string
+        country: country (Scotland, England or Wales)
+
+    Retruns:
+        Updated construction age band
+    """
+    if construction_age_band == "unknown":
+        if transaction_type == "new dwelling":
+            if ~pd.isnull(inspection_date) and (inspection_date is not None):
+                # inspection_year = datetime.strptime(inspection_date, "%Y-%m-%d").year
+                inspection_year = inspection_date.year
+                if inspection_year >= 2007:
+                    return "2007 onwards"
+                elif inspection_year >= 2003:
+                    return "2003-2007"
+                elif inspection_year >= 1996:
+                    return "1996-2002"
+                elif inspection_year >= 1991:
+                    return "1991-1998"
+                elif inspection_year >= 1983:
+                    return "1983-1991"
+                elif inspection_year >= 1976:
+                    return "1976-1983"
+                elif inspection_year >= 1965:
+                    return "1965-1975"
+                elif inspection_year >= 1950:
+                    return "1950-1966"
+                elif inspection_year >= 1930:
+                    return "1930-1949"
+                elif inspection_year >= 1919:
+                    return "1900-1929"
+                elif (inspection_year >= 1900) and (country != "Scotland"):
+                    return "1900-1929"
+                elif (inspection_year < 1900) and (country != "Scotland"):
+                    return "England and Wales: before 1900"
+                elif (inspection_year <= 1919) and (country == "Scotland"):
+                    return "Scotland: before 1919"
```

Comment:
EPC inspections only began in 2008, so this isn't needed - if it's a new dwelling then the age band will be "2007 onwards"

Reply:
ahhhh of course, you're absolutely right!